exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +79,245 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+				UID:       "owner-mp-uid-1",
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "MachinePool",
+				APIVersion: expclusterv1.GroupVersion.String(),
+			},
+			Spec: expclusterv1.MachinePoolSpec{
+				ClusterName: ownerCluster.Name,
+				Template: clusterv1.MachineTemplateSpec{
+					Spec: clusterv1.MachineSpec{
+						ClusterName: ownerCluster.Name,
+					},
+				},
+			},
+		}
+
+		rosaMachinePool = &expinfrav1.ROSAMachinePool{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "rosa-machinepool",
+				Namespace: ns.Name,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						Name:       ownerMachinePool.Name,
+						UID:        ownerMachinePool.UID,
+						Kind:       "MachinePool",
+						APIVersion: clusterv1.GroupVersion.String(),
+					},
+				},
+			},
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ROSAMachinePool",
+				APIVersion: expinfrav1.GroupVersion.String(),
+			},
+			Spec: expinfrav1.RosaMachinePoolSpec{},
+		}
+
+		objects = []client.Object{secret, ownerCluster, ownerMachinePool}
+
+		for _, obj := range objects {
+			createObject(g, obj, ns.Name)
+		}
+	}
+
+	teardown := func() {
+		mockCtrl.Finish()
+		for _, obj := range objects {
+			cleanupObject(g, obj)
+		}
+	}
+
+	t.Run("Reconcile create node pool", func(t *testing.T) {
+		setup(t)
+		defer teardown()
+		ocmMock = mocks.NewMockOCMClient(mockCtrl)
+		expect := func(m *mocks.MockOCMClientMockRecorder) {
+			m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+				return nil, false, nil
+			}).Times(1)
+			m.CreateNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePool *cmv1.NodePool) (*cmv1.NodePool, error) {
+				return nodePool, nil
+			}).Times(1)
+		}
+		expect(ocmMock.EXPECT())
+
+		c := fake.NewClientBuilder().WithScheme(scheme).WithObjects(rosaMachinePool, ownerCluster, ownerMachinePool, rosaControlPlane, secret).Build()
+		stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+		stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Times(1)
+
+		r := ROSAMachinePoolReconciler{
+			Recorder:         recorder,
+			WatchFilterValue: "",
+			Endpoints:        []scope.ServiceEndpoint{},
+			Client:           c,
+			NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+			NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+				return ocmMock, nil
+			},
+		}
+
+		req := ctrl.Request{}
+		req.NamespacedName = types.NamespacedName{Name: "rosa-machinepool", Namespace: ns.Name}
+
+		result, err := r.Reconcile(ctx, req)
+
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(result).To(Equal(ctrl.Result{}))
+	})
+
+	t.Run("Reconcile delete", func(t *testing.T) {
+		setup(t)
+		defer teardown()
+
+		deleteTime := metav1.NewTime(time.Now().Add(5 * time.Second))
+		rosaMachinePool.ObjectMeta.Finalizers = []string{"finalizer-rosa"}
```

Comment:
Not sure what are you testing here, but finalizer should be added by the RosaMachinePool controller

Reply:
Are you sure? I think it should be removed on delete https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/exp/controllers/rosamachinepool_controller.go#L318